concordium-consensus/src/Concordium/MultiVersion.hs

```diff
@@ -1715,13 +1716,31 @@ receiveFinalizationRecord gi finRecBS = withLatestExpectedVersion_ gi $ \case
     (EVersionedConfigurationV0 (vc :: VersionedConfigurationV0 finconf pv)) ->
         case runGet getExactVersionedFinalizationRecord finRecBS of
             Left err -> do
-                logEvent Runner LLDebug $ "Could not deserialized finalization record: " ++ err
+                logEvent Runner LLDebug $ "Could not deserialize finalization record: " ++ err
                 return Skov.ResultSerializationFail
             Right finRec -> runSkovV0Transaction vc (finalizationReceiveRecord False finRec)
     (EVersionedConfigurationV1 _) -> do
         logEvent Runner LLDebug "Unexpected finalization record event in consensus version 1."
         return Skov.ResultInvalid
 
+receiveFinalizationEntry :: GenesisIndex -> ByteString -> MVR finconf Skov.UpdateResult
+receiveFinalizationEntry gi finEntryBS =
+    withLatestExpectedVersion_ gi $ \case
+        (EVersionedConfigurationV0 _) -> do
+            logEvent Runner LLDebug "Unexpected finalization entry event in consensus version 0."
+            return Skov.ResultInvalid
+        (EVersionedConfigurationV1 (vc :: VersionedConfigurationV1 finconf pv)) -> do
+            case runGet get finEntryBS of
+                Left err -> do
+                    logEvent Runner LLDebug $ "Could not deserialize finalization entry: " <> err
+                    return Skov.ResultSerializationFail
+                Right finEntry -> do
+                    runSkovV1Transaction vc (SkovV1.catchupFinalizationEntry finEntry) >>= \case
+                        SkovV1.CFERSuccess -> return Skov.ResultSuccess
+                        SkovV1.CFERInconsistent -> return Skov.ResultUnverifiable
+                        SkovV1.CFERInvalid -> return Skov.ResultUnverifiable
```

Comment:
Why is invalid mapped to unverifiable here?

Reply:
`ResultInvalid` is probably the better choice yes.